src/lib/users/ssh_authorized_keyring.rb

```diff
@@ -139,71 +142,34 @@ def authorized_keys_path(home)
       # should support changing SSH_DIR to something like `.config/ssh`.
       #
       # @param home  [String] Home directory where SSH directory must be created
-      # @param owner [Fixnum] Owner's UID
-      # @param group [Fixnum] Owner's GID
+      # @param user  [Fixnum] Users's UID
+      # @param group [Fixnum] Group's GID
       # @return [String] Returns the path to the first created directory
-      def create_ssh_dir(home, owner, group)
+      #
+      # @raise SSHDirectoryIsLink
+      # @raise CouldNotCreateSSHDirectory
+      def create_ssh_dir(home, user, group)
         ssh_dir = ssh_dir_path(home)
+        raise SSHDirectoryIsLink if FileUtils::IsLink(ssh_dir)
```

Comment:
NP: Maybe we should invert the condition: `raise ... if FileUtils::Exists(ssh_dir) && !FileUtils::IsDir(ssh_dir)` to handle also other non-dir cases.